daml-script/test/daml/upgrades/DataChanges.daml

```diff
@@ -173,112 +173,113 @@ templateNonOptionalFieldsAddedNested =
     V2.NonOptionalFieldsAddedNestedCall
 
 {- PACKAGE
-name: data-changes-variant-unchanged
+name: data-changes-enum-unchanged
 versions: 2
 modules:
-  - name: VariantUnchanged
+  - name: EnumUnchanged
     contents: |
-      data VariantUnchangedData
-        = VariantUnchangedData1
-        | VariantUnchangedData2
-        deriving (Eq, Show)
-      template VariantUnchanged
+      data EnumUnchangedData
+        = EnumUnchangedData1
+        | EnumUnchangedData2
+        deriving (Eq, Show, Enum)
+      template EnumUnchanged
         with
           party : Party
-          nestedData : VariantUnchangedData
+          nestedData : EnumUnchangedData
         where
           signatory party
-          choice VariantUnchangedCall : Text
+          choice EnumUnchangedCall : Text
             controller party
             do pure V1("V1")V2("V2")
 -}
 
-templateVariantUnchanged : Script ()
-templateVariantUnchanged =
+templateEnumUnchanged : Script ()
+templateEnumUnchanged =
   templateInvalidChange
-    @V2.VariantUnchanged
+    @V2.EnumUnchanged
     True
-    (`V1.VariantUnchanged` V1.VariantUnchangedData1)
-    V2.VariantUnchangedCall
+    (`V1.EnumUnchanged` V1.EnumUnchangedData1)
+    V2.EnumUnchangedCall
 
 {- PACKAGE
-name: data-changes-variant-removed
+name: data-changes-enum-removed
 versions: 2
 modules:
-  - name: VariantRemoved
+  - name: EnumRemoved
     contents: |
-      data VariantRemovedData
-        = VariantRemovedData1
-        V1(| VariantRemovedData2)
-        deriving (Eq, Show)
-      template VariantRemoved
+      data EnumRemovedData
+        = EnumRemovedData1
+        | EnumRemovedData2
+        V1(| EnumRemovedData3)
```

Comment:
these are all, strictly speaking, enums and not variants, since all constructors are nullary, so I renamed them. I also added `Enum` to the derived instances to show this, since it wouldn't be possible to derive it for non-enum variants [which is how I found #18504, the reason for adding `EnumRemovedData3`]